x-pack/plugins/profiling/public/components/differential_topn_functions_grid/index.tsx

```diff
@@ -0,0 +1,492 @@
+                setSelectedFrameId({ currentFrameId, isComparison });
+              }}
+              iconType="inspect"
+            >
+              {i18n.translate('xpack.profiling.compareFrame.component.findLabel', {
+                defaultMessage: 'Find corresponding frame',
+              })}
+            </Component>
+          }
+          isOpen={isOpen}
+          closePopover={() => {
+            setSelectedFrameId(undefined);
+          }}
+          anchorPosition="upRight"
+          css={css`
+            .euiPopover__anchor {
+              align-items: start;
+              display: flex;
+            }
+          `}
+        >
+          {compareRow ? (
+            <div style={{ maxWidth: 400 }}>
+              <EuiPopoverTitle paddingSize="s">
+                {isComparison
+                  ? i18n.translate('xpack.profiling.diffTopNFunctions.baseLineFunction', {
+                      defaultMessage: 'Baseline function',
+                    })
+                  : i18n.translate('xpack.profiling.diffTopNFunctions.comparisonLineFunction', {
+                      defaultMessage: 'Comparison function',
+                    })}
+              </EuiPopoverTitle>
+              <EuiTitle size="xs">
+                <EuiText>{getCalleeFunction(compareRow.frame)}</EuiText>
+              </EuiTitle>
+              <EuiBasicTable
+                items={[compareRow]}
+                columns={[
+                  { field: 'rank', name: 'Rank' },
+                  {
+                    field: 'samples',
+                    name: 'Samples',
+                    render: (_, value) => value.samples.toLocaleString(),
+                  },
+                  {
+                    field: 'selfCPUPerc',
+                    name: 'Self CPU',
+                    render: (_, value) => `${value.selfCPUPerc.toFixed(2)}%`,
+                  },
+                  {
+                    field: 'totalCPUPerc',
+                    name: 'Total CPU',
+                    render: (_, value) => `${value.totalCPUPerc.toFixed(2)}%`,
+                  },
+                ]}
+              />
+            </div>
+          ) : (
+            <EuiText color="subdued" size="s">
+              {i18n.translate('xpack.profiling.diffTopNFunctions.noCorrespondingValueFound', {
+                defaultMessage: 'No corresponding value found',
+              })}
+            </EuiText>
+          )}
+        </EuiPopover>
+      );
+    },
+    [baseRows, comparisonRows, selectedFrameId]
+  );
+
+  const sortedBaseRows = useMemo(() => {
+    return sortRows(baseRows, sortField, sortDirection);
+  }, [baseRows, sortDirection, sortField]);
+
+  const sortedComparisonRows = useMemo(() => {
+    return sortRows(
+      comparisonRows,
+      removeComparisonFromId(comparisonSortField),
+      comparisonSortDirection
+    );
+  }, [comparisonRows, comparisonSortDirection, comparisonSortField]);
+
+  const columns: EuiDataGridColumn[] = useMemo(
```

Comment:
We could move this and the `compareFrameAction` to another file, hook, or just outside the component. Wdyt? It would make the main `DifferentialTopNFunctionsGrid` smaller and easier to read.

Reply:
I moved the columns to another file. But I left the `compareFrameAction` within the component as needs access to the state.

Reply:
ah, I changed my mind and moved the `compareFrameAction` to its own file in the end. Thanks for the suggestion.